libcst/codegen/gen_matcher_classes.py

```diff
@@ -0,0 +1,528 @@
+        typestr = typestr[8:-2]
+
+    # Now, parse the expression with LibCST.
+    cleanser = CleanseFullTypeNames()
+    typecst = parse_expression(typestr)
+    typecst = typecst.visit(cleanser)
+    clean_type: Optional[cst.CSTNode] = None
+
+    # Now, convert the type to allow for DoNotCareSentinel values.
+    if isinstance(typecst, cst.Subscript):
+        if typecst.value.deep_equals(cst.Name("Union")):
+            # We can modify this as-is to add our type
+            clean_type = typecst.with_changes(
+                slice=[*typecst.slice, _get_do_not_care()]
+            )
+        elif typecst.value.deep_equals(cst.Name("Literal")):
+            clean_type = _get_wrapped_union_type(typecst, _get_do_not_care())
+        elif typecst.value.deep_equals(cst.Name("Sequence")):
+            clean_type = _get_wrapped_union_type(typecst, _get_do_not_care())
+
+    elif isinstance(typecst, (cst.Name, cst.SimpleString)):
+        clean_type = _get_wrapped_union_type(typecst, _get_do_not_care())
+
+    # Now, clean up the outputted type and return the code it generates. If
+    # for some reason we encounter a new node type, raise so we can triage.
+    if clean_type is None:
+        raise Exception(f"Don't support {typecst}")
+    else:
+        # First, add DoNotCareSentinel to all sequences, so that a sequence
+        # can be defined partially with explicit DoNotCare() values for some
+        # slots.
+        clean_type = ensure_type(
+            clean_type.visit(AddDoNotCareToSequences()), cst.CSTNode
+        )
+        # Now, double-quote any types we parsed and repr'd, for consistency.
+        clean_type = ensure_type(clean_type.visit(DoubleQuoteStrings()), cst.CSTNode)
+        # Now, insert OneOf/AllOf and MatchIfTrue into unions so we can typecheck their usage.
+        # This allows us to put OneOf[SomeType] or MatchIfTrue[cst.SomeType] into any
+        # spot that we would have originally allowed a SomeType.
+        clean_type = ensure_type(
+            clean_type.visit(AddLogicAndLambdaMatcherToUnions()), cst.CSTNode
+        )
+        # Now, insert AtMostN and AtLeastN into sequence unions, so we can typecheck
+        # them. This relies on the previous OneOf/AllOf insertion to ensure that all
+        # sequences we care about are Sequence[Union[<x>]].
+        clean_type = ensure_type(
+            clean_type.visit(AddWildcardsToSequenceUnions()), cst.CSTNode
+        )
+        # Finally, generate the code given a default Module so we can spit it out.
+        return cst.Module(body=()).code_for_node(clean_type)
+
+
+@dataclass(frozen=True)
+class Field:
+    name: str
+    type: str
+
+
+def _get_fields(node: Type[cst.CSTNode]) -> Generator[Field, None, None]:
+    """
+    Given a CSTNode, generate a field name and type string for each.
+    """
+
+    for field in fields(node) or []:
+        if field.name == "_metadata":
+            continue
```

Comment:
Based on the test report, this never heppend https://codecov.io/gh/Instagram/LibCST/pull/75/diff?src=pr&el=tree#D6-577

Reply:
Its dependent on Python version from my testing.